File: rxbinding/src/main/java/com/forceson/rxbinding/widget/RxSeekBar.java
package com.forceson.rxbinding.widget;

import android.widget.SeekBar;

import rx.Observable;

/**
 * Created by son on 2020-01-25.
 */
public class RxSeekBar {
    public static Observable<Integer> changes(SeekBar view) {
        return Observable.create(new SeekBarChangeObservable(view));
    }

    public static Observable<SeekBarChangeEvent> changeEvents(SeekBar view) {
        return Observable.create(new SeekBarChangeEventObservable(view));
    }

    private RxSeekBar() {
        throw new AssertionError("No instances.");
    }
}

File: rxbinding-kotlin/src/main/java/com/forceson/rxbinding/kotlin/widget/RxSeekBar.kt
package com.forceson.rxbinding.kotlin.widget

import android.widget.SeekBar
import com.forceson.rxbinding.widget.RxSeekBar
import com.forceson.rxbinding.widget.SeekBarChangeEvent
import rx.Observable

/**
 * Created by son on 2020-02-10.
 */

fun SeekBar.changes(): Observable<Int> = RxSeekBar.changes(this)

fun SeekBar.changeEvents(): Observable<SeekBarChangeEvent> = RxSeekBar.changeEvents(this)
